### PR TITLE
edges are not bound-able

### LINKED ISSUE #465: Crash encountered about same relationship variable in pattern
ref: https://github.com/RedisGraph/RedisGraph/issues/3206

I execute a query on my empty RedisGraph and get a crash.

Redis version:7.2.0
RedisGraph version: v2.12.10
operating system: windows 11
API：Cypher version 9
query:WITH NULL AS a0 MATCH ()-[a0]-()-[a0]-() RETURN 1
error message:"Reached the max retries per request limit (which is 1). Refer to "maxRetriesPerRequest" option for details."

Actual behavior:
The query failed and the redisgraph container exited abnormally with the error message.

### PR DESCRIPTION
**Pull request by swilly22**
## **User description**
Resolves #465

Unlike nodes, edges always declare alias
Example:
```cypher
MATCH (a) WITH a MATCH (z)-[]->(a) RETURN z
```
Here `a` is declared by the first `MATCH` clause and `a` is considered bounded by the second `MATCH` clause.

This behavior isn't applicable for edges
```cypher
MATCH ()-[e]->() WITH e MATCH ()-[e]->() RETURN *
```
In this case `e` is declared by **BOTH** `MATCH` clauses, which is not allowed.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved validation logic for `CREATE` and `MERGE` clauses to handle relation types, directions, and properties accurately.
  - Refined error messages and handling for better clarity during graph pattern creation.

- **Tests**
  - Added extra lines and comments for clarity in various test methods.
  - Updated test cases to handle redeclaration errors for bound edges and variables.
  - Enhanced exception handling in `test07_null_graph_entity_inputs` to manage specific error scenarios during query execution.

___

## **CodeAnt-AI Description**
Reject redeclaration of bound edges in CREATE/MERGE and enforce relationship constraints

### What Changed
- CREATE and MERGE now reject reusing an already-bound relationship alias (edges). Queries that redeclare the same edge variable now raise an error like "The bound edge 'e' can't be redeclared in a CREATE/MERGE clause".
- CREATE/MERGE now require exactly one relationship type, require the relationship to be directed, and disallow variable-length relationship patterns; violations produce clause-specific error messages.
- Test suites updated to expect the new edge-specific error wording and include a new test covering bound-edge redeclaration scenarios.

### Impact
`✅ Clearer redeclare errors for bound edges`
`✅ Prevent duplicate edge declarations in CREATE/MERGE`
`✅ Reject undirected or variable-length relationships in CREATE/MERGE`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: src/ast/ast_validations.c [hunk 1]
```diff
@@ -792,67 +736,61 @@ static VISITOR_STRATEGY _Validate_rel_pattern
...
-		if(alias_type == raxNotFound) {
-			_IdentifierAdd(vctx, alias, (void*)T_EDGE);
-			return VISITOR_RECURSE;
-		}
-			
-		if(alias_type != (void *)T_EDGE && alias_type != NULL) {
-			ErrorCtx_SetError(EMSG_SAME_ALIAS_NODE_RELATIONSHIP, alias);
-			return VISITOR_BREAK;
-		}
+		void *alias_type  = _IdentifiersFind(vctx, alias);
```
Comment (AviAvni) [inline src/ast/ast_validations.c:789]: is this needed?
